### DISCUSSION TITLE
Testing with Iceoryx in the loop

### DISCUSSION THREAD

**Discussion by forestf255** [eclipse-iceoryx/iceoryx#2045] [Q&A] [answered]
Hey all,

I'm looking for any recommendations on testing once Iceoryx has been integrated into a project. 

Since Iceoryx/Roudi relies on some compile-time set filepaths and the [IceOryxRouDiMemoryManager](https://github.com/eclipse-iceoryx/iceoryx/blob/v2.0.3/iceoryx_posh/include/iceoryx_posh/roudi/memory/iceoryx_roudi_memory_manager.hpp#L62-L80) disallows running multiple instances of Roudi at once, it seems it is not possible to run integration tests in parallel that depend on Iceoryx. Are there any options to run Roudi fully encapsulated within a process without it writing to the file system?

Thanks!

**Comment by elBoberido** [comment, marked as answer]
@forestf255 no, currently this is not possible. There are some ideas to add something like a domain ID to RouDi but there is no ETA on this.
